Sandbox branch

joshua-git0 opened the pull request:
Resolves SJSURoboticsTeam/urc-central-2021#24

Commit:
sandbox foo sandbox

reslolves issue #1
:

Add README to Control Systems 2021

Resolves SJSURoboticsTeam/urc-central-2021#22

AndrewDEmerson committed:
added brushed dc inital draft

AndrewDEmerson committed:
Made requested changes to motor interface. SJSURoboticsTeam/urc-central-2021#26

joshua-git0 committed:
Added rotational encoder interface

joshua-git0 committed:
rotational encoder interface

AndrewDEmerson committed:
returns correct error types SJSURoboticsTeam/urc-central-2021#26

AndrewDEmerson committed:
added .clang-format file so vscode will auto format files

kammce commented:
@josh-github1 can you rename the title of this project. Also, you need to associate your commit with the issue number in central.

```
Resolves SJSURoboticsTeam/urc-central-2021/issues/#24
```

To your commit message or at least to your PR. so edit the PR description to include the message above.

kammce commented on .clang-format:
```diff
@@ -0,0 +1,35 @@
+BasedOnStyle: Google
```
Remove clang from this PR. You will need to rebase your branch from master as it already has clang format in it.

joshua-git0 replied:
I have a lot to learn.. I just looked up git rebasing

kammce replied:
This is still in here.

kammce commented on .gitignore:
```diff
@@ -0,0 +1,45 @@
+projects/*
+projects/**/build
+
+!projects/barebones
+!projects/hello_world
+!projects/starter
+
+demos/**/build
+library/static_libraries/*
+!library/static_libraries/keep_folder
+
+tools/**/*
+!tools/build_all_projects.sh
+!tools/cpplint
+!tools/FlashMagic
+!tools/Hercules.exe
+!tools/link_projects.sh
+!tools/makefile_help_menu.txt
+!tools/presubmit.sh
+!tools/putty32.exe
+!tools/putty64.exe
+!tools/launch_openocd_gdb.sh
+!tools/generate_doxygen_docs.sh
+!tools/doxygen_ignore_list.txt
+!tools/common.sh
+!tools/find_sources_without_tests.sh
+!tools/print_unused_test_files.sh
+!tools/gdb_dashboard
+!spell_checker.sh
+
+api/doxygen_stderr.txt
```
you can remove from your PR. Add it as a new PR.

kammce replied:
This is still in here.

kammce commented on Common/RotationalEncoder.hpp:
```diff
@@ -0,0 +1,19 @@
+  virtual Returns<void> Enable() = 0;
+  virtual void SetAngleToZero()  = 0;
+}
+}  // namespace sjsu
```
Add a new line at the end of this file.

kammce commented on Common/RotationalEncoder.hpp:
```diff
@@ -0,0 +1,19 @@
+{
+ public:
+  virtual Returns<void> Initialize()            = 0;
+  virtual Returns<units::degrees> GetRotation() = 0;
```
This line will not compile. Test this out in SJSU-Dev2 if you need to but you will see that it does not work.

kammce commented on Common/RotationalEncoder.hpp:
```diff
@@ -0,0 +1,19 @@
+#include <cstdint>
+
+#include "utility/status.hpp"
+
```
Your are going to need the "utility/units.hpp" library included here as well.

kammce commented on Common/RotationalEncoder.hpp:
```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cmath>
+#include <cstdint>
```
I don't see you using cmath or cstdint in this file. Maybe you can remove these.

kammce commented on README.md:
```diff
@@ -0,0 +1,3 @@
+# SJSU Robotics Control Systems 2021
+
+Add more content later.
```
Not sure why this file is in here. You may need to pull the latest master branch, then rebase off of that.

AndrewDEmerson committed:
Updated MPU6050 accelometer class (#8)

* Rewrote MPU6050 class for new interface

kammce reviewed (changes requested):
You haven't resolved any of my comments.

AndrewDEmerson committed:
Add MPU6050 Demo

Resolves SJSURoboticsTeam/urc-central-2021#84

AndrewDEmerson committed:
Fixed Incorrect Bit Manipulation in MPU6050 (#18)

Fixed the issue with the bit masking by using the bit manipulation library.
Fixed another issue where the MPU was put in sleep mode before setting the full scale range, causing it to not update the value.

Resolves #14

naterpotatoers committed:
adds clone and branch info to readme (#20)

naterpotatoers committed:
Initial template_application (#10)

* adds demo & walk through of two SJTwo LED tasks using SJDev2 rtos class

kammce committed:
Add Multiplatform RMD X Tech Demo (#22)

Add .gitignore file that excludes */build directories from the repo as
these folders usually contain binaries and other temp files that are not
needed in the main repo.

Resolves SJSURoboticsTeam/urc-central-2021#9
Resolves SJSURoboticsTeam/urc-central-2021#8

Co-authored-by: Khalil Estell <kammce@google.com>

AndrewDEmerson committed:
Arm control prototype (#24)

* Add initial Arm Class.

naterpotatoers committed:
Drive class scaffolder (#23)

* adds initial Rover drive system class

* adds initial Rover wheel class

* adds comments to function stubs

* fixes string compilation error

* moves old Arm files to Old dir

* adds drive spin mode file to Demos dir

naterpotatoers committed:
Initial drive modes implementation (#25)

* adds mission control data struct

* adds initial drive mode setters and handlers

* adds better comments and adds initial rover data functions/code

* adds clamp TODO comment

* fixes GitHub comments - dead code & removes functions

Commit:
added some content to drive main

Commit:
drive content

naterpotatoers commented:
going to close this PR since I don't think it will ever get merged
